src/HubManagerInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+use Sentry\State\HubInterface;
+
+interface HubManagerInterface
```

Comment:
Out of curiosity, the hub manager is here to act as a Singleton right? So why defining an interface for it?

His role is to allow access to the global instance of the HubInterface and that's all no?

Reply:
Actually it's a singleton because it's a leftover of some tries I did to find a solution that would satisfy the requirements I would like to meet for this PR to be ready for review and that I wrote in the first comment. In particular I would like to be able to support getting the hub manager from a service container, so it should not be a singleton and this is the reason I "migrated" to hold the instance in the global function `getHubManager`. However, this solution has drawbacks too:

- Once the variable is initialized you have to have a way to replace it in case someone called the function too early and you don't have control on this (this is the purpose of the `$replace` parameter)
- Having a singleton (this doesn't really depends on where the singleton is placed, but of course it's even worse if it's in a global function) causes problems in the unit tests where the state of the "application" is shared between them
- Singletons are generally speaking a bad idea in all cases, from testing to usage

Unfortunately supporting API based on global functions means that we need the singleton, no way around this. Another solution would probably be having a factory that holds the singleton so that neither `getHubManager` nor the hub manager itself holds the instance, but I didn't investigate if this idea can work and/or makes sense. WDYT?

Reply:
IMHO, only one and one instance could exist at a time and this instance should be created by calling `Sentry\init()` explicitly.

Trying to magically instanciate the Hub is pointless as the SDK cannot guess the whole configuration wanted by the user. Trying to get some of them from the environment variable is a bad idea as it is a global state that you don't want to deal with in the library. 

Static method are ok to be used in service container so that:

- It stays ok to be used standalone without FW / service container
- It can ensure only once instance exist at a time
- It acts as the only way to configure / get a Hub

Reply:
I've changed the design to be more similar to what the C# SDK does, which basically is having a class that acts as main entry point and that holds the instance of the hub, and if someone wants to use the DI pattern then he can register in the container the `HubAdapter` which acts as a sort of proxy between the user and the class `SentrySdk` so that you can typehint against the `HubInterface` instead of using directly that singleton. I've also deprecated the global functions, I'm not sure this is something we want to do but it would simplify the whole code a bit given that they are now just duplicates of the methods of the `SentrySdk` class